src/openms/source/FORMAT/PepXMLFile.cpp

```diff
@@ -804,22 +913,81 @@ namespace OpenMS
       // may appear to be "out of bounds" - see NOTE above:
       String identifier = current_proteins_[min(UInt(current_proteins_.size()), search_id_) - 1]->getIdentifier();
       current_peptide_.setIdentifier(identifier);
+
+      // set optional attributes
+      if (!native_spectrum_name_.empty()) 
+      {
+        current_peptide_.setMetaValue("pepxml_spectrum_name", native_spectrum_name_);
```

Comment:
this leads to a few testfailues and a `UserParam` to be present in every identifcation in the idXML file. Maybe we can discuss whether we really want this or not, we might also skip it and not keep that information around